assets/src/edit-story/components/panels/layer/useLayerReordering.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import {
+  useState,
+  useCallback,
+  useEffect,
+  useContext,
+  useRef,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import LayerContext from './context';
+
+function useLayerReordering(layer) {
+  const { type, id: elementId, position: currentPosition } = layer;
+
+  const isBackground = type === 'background';
+
+  const [dragTarget, setDragTarget] = useState(null);
+  const {
+    state: { currentSeparator },
+    actions: { setIsReordering, setCurrentSeparator },
+  } = useContext(LayerContext);
+
+  const {
+    actions: { arrangeElement, setSelectedElementsById },
+  } = useStory();
+
+  const handleStartReordering = useCallback(
+    (evt) => {
+      if (!isBackground && !evt.shiftKey && !evt.metaKey) {
```

Comment:
This line could do with a comment explaining what is going on here.

Reply:
Fixed in a248408.

Reply:
Optimized a bit in def6b30d33be2c8b85c548ffc67ed1dd9b3caf09.

Reply:
Ah, thanks! Got it!